wdtk-dumpfiles/src/main/java/org/wikidata/wdtk/dumpfiles/wmf/JsonOnlineDumpFile.java

```diff
@@ -115,9 +120,30 @@
 
 	@Override
 	protected boolean fetchIsDone() {
-		// WMF provides no easy way to check this for these files;
-		// so just assume it is done
-		return true;
+		boolean result = false;
+		String fileName = WmfDumpFile.getDumpFileName(DumpContentType.JSON,
+				this.projectName, this.dateStamp);
+		try (InputStream in = this.webResourceFetcher
+				.getInputStreamForUrl(getBaseUrl())) {
+			BufferedReader bufferedReader = new BufferedReader(
+					new InputStreamReader(in, StandardCharsets.UTF_8));
+			String inputLine;
+			while ((inputLine = bufferedReader.readLine()) != null) {
+				System.out.println(inputLine);
+				if (inputLine.startsWith("<a href=")) {
+					System.out.println(inputLine);
+					inputLine = inputLine.replaceAll(".*href=\"(.*?)\".*", "$1");
```

Comment:
I am worried that this way of parsing the HTML file is rather brittle: it relies on the specifics of how the HTML content is formatted, which should normally be irrelevant (it could well change if WMF upgrades the underlying HTTP server).
What value does it add beyond the `return true` that was there?
If `return true` is not satisfactory, what do you think of making a HEAD request to the dump URL to check if it exists? I think that would be more reliable.

Reply:
I copied what was done in another place in the code but yes a head request is a lot more robust (and a bit faster too), I'll change that